web/taskmaster/src/app/components/loginform/loginform.component.ts

```diff
@@ -23,20 +26,36 @@ import { LoginData } from '../../models/logindata.model';
   styleUrl: './loginform.component.scss',
 })
 export class LoginformComponent {
-  @Input() disabled = false;
-
-  @Output() loginEvent = new EventEmitter<LoginData>();
+  disabled = false;
 
   loginForm = new FormGroup({
     username: new FormControl<string>('', Validators.required),
     password: new FormControl<string>('', Validators.required),
   });
 
+  constructor(
+    private loginService: LoginService,
+    private snackBarService: SnackBarService,
+  ) {}
+
   login() {
-    const loginData = {
-      username: this.loginForm.value.username!,
-      password: this.loginForm.value.password!,
-    };
-    this.loginEvent.emit(loginData);
+    const loginData = new LoginData(
+      this.loginForm.value.username!,
+      this.loginForm.value.password!
+    );
+    this.disabled = true;
+    this.loginService.login(loginData).pipe(
+    finalize(() => {
+      this.disabled = false;
+    }),
+    catchError(error => {
+      this.snackBarService.openSnackbar('Login Failed', 'error');
+      return EMPTY;
+    })).subscribe((jwt) => {
```

Comment:
catchError is useful when you make error handling other than just triggering the snackbar. 

For this situation is better to have
```
.subscribe({
  next: (jwt) => {
      localStorage.setItem('jwt', jwt.access_token);
      localStorage.setItem('refresh', jwt.refresh_token);
      this.loginService.updateStatus(true);
      this.snackBarService.openSnackbar('Login Successful', 'success');
  },
  error: () =>  this.snackBarService.openSnackbar('Login Failed', 'error');
 })
```